java/src/test/java/org/rocksdb/SstFileWriterTest.java

```diff
@@ -0,0 +1,81 @@
+package org.rocksdb;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.rocksdb.util.BytewiseComparator;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SstFileWriterTest {
+    private static final String SST_FILE_NAME = "test.sst";
+    private static final String DB_DIRECTORY_NAME = "test_db";
+
+    @ClassRule
+    public static final RocksMemoryResource rocksMemoryResource =
+            new RocksMemoryResource();
+
+    @Rule
+    public TemporaryFolder parentFolder = new TemporaryFolder();
+
+    private File newSstFile(TreeMap<String, String> keyValues) throws IOException, RocksDBException {
+        EnvOptions envOptions = new EnvOptions();
+        SstFileWriter sstFileWriter = null;
+        ComparatorOptions comparatorOptions = new ComparatorOptions();
+        BytewiseComparator comparator = new BytewiseComparator(comparatorOptions);
+        Options options = new Options().setComparator(comparator);
+        File sstFile = parentFolder.newFile(SST_FILE_NAME);
+        try {
+            sstFileWriter = new SstFileWriter(envOptions, options, comparator);
+            sstFileWriter.open(sstFile.getAbsolutePath());
+            for (Map.Entry<String, String> keyValue : keyValues.entrySet()) {
+                Slice keySlice = new Slice(keyValue.getKey());
+                Slice valueSlice = new Slice(keyValue.getValue());
+                sstFileWriter.add(keySlice, valueSlice);
+                keySlice.close();
+                valueSlice.close();
+            }
+            sstFileWriter.finish();
+        } finally {
+            assertThat(sstFileWriter).isNotNull();
+            sstFileWriter.close();
+            options.close();
+            envOptions.close();
+            comparatorOptions.close();
+            comparator.close();
+        }
+        return sstFile;
+    }
+
+    @Test
+    public void generateSstFile() throws RocksDBException, IOException {
+        TreeMap<String, String> keyValues = new TreeMap<>();
+        keyValues.put("key1", "value1");
+        keyValues.put("key2", "value2");
+        newSstFile(keyValues);
+    }
+
+    @Test
+    public void ingestSstFile() throws RocksDBException, IOException {
+        TreeMap<String, String> keyValues = new TreeMap<>();
```

Comment:
Please use `final` where you don't mutate a variable. Also should apply this above...